.github/workflows/backend-api-push-to-main.yml

```diff
@@ -54,7 +54,8 @@ jobs:
     runs-on: ubuntu-latest
```

Comment:
I think these Dev and Build `push-and-test-image` should be in separate jobs. I can see the the Dev `BACKEND_API_DEV_COSIGN_KEY` has been deleted which would be needed for to sign the image in Dev, though this is currently commented out

Reply:
Happy to do this. I did this to reduce the workflow time and avoid caching the docker image + sam application (which is possible, but adds a bit more complexity)

The reason I think it would be good to separate it would be because we could run the jobs in parallel and they shouldn't depend on each other given Dev is not part of our route to live.

What's your rationale for splitting it out, out of curiosity?

Reply:
Because it is an already established pattern within Mobile Pod. 😅 

It will help us identify when and where this GitHub pipeline breaks more easily and only re-run jobs that have failures.

I am unsure if there are any form of implications (security wise) from using both Dev and Build environment variables in the same job but seems cleaner to me to separate them out.

Might be a better question for @Pranjal-GDS as to why we have chosen this method?